src/circuits/mimc/mimc_hash.hpp

```diff
@@ -0,0 +1,42 @@
+// DISCLAIMER:
+// Content taken and adapted from:
+// https://github.com/HarryR/ethsnarks/blob/master/src/gadgets/mimc.hpp
+
+#ifndef __ZETH_MIMC_HASH_HPP__
+#define __ZETH_MIMC_HASH_HPP__
+
+#include "snarks_alias.hpp"
+#include "mimc.hpp"
+
+namespace libzeth {
+
+template<typename FieldT>
+class MiMC_hash_gadget:public libsnark::gadget<FieldT> {
+/*
+  MiMC_hash_gadget enforces correct computation of a MiMCHash based on a MiMC permutation with exponent 7
+  */
+public:
+	std::vector<MiMCe7_permutation_gadget<FieldT>> permutation_gadgets; // vector of permutation gadgets
+	std::vector<libsnark::pb_variable<FieldT>> messages;  //  vector of messages to process
+	libsnark::pb_variable_array<FieldT> outputs; // vector of round outputs variables
+	const libsnark::pb_variable<FieldT> iv; // initial vector variable
+  const libsnark::pb_variable<FieldT> out; // expected output variable
```

Comment:
Tab

Reply:
Fixed

Reply:
It's weird because if I open the file online it shows it me with the tab. but here still it is without the tab.